### PR TITLE
Unbreak containers without images

### PR DESCRIPTION
**Pull request by nalind**
Make it possible to create a container without using an underlying image.

### CODE REVIEW

File: storage/store.go [hunk 1]
```diff
@@ -1369,7 +1375,9 @@ func (s *store) DeleteImage(id string, commit bool) (layers []string, err error)
 		}
 		aContainerByImage := make(map[string]string)
 		for _, container := range containers {
-			aContainerByImage[container.ImageID] = container.ID
+			if container.ImageID != "" {
+				aContainerByImage[container.ImageID] = container.ID
```
Comment (rhatdan) [inline storage/store.go]: Is it ok if this is not set?  Or should it cause an error?

Comment (rhatdan) [inline storage/store.go]: Otherwise LGTM

Comment (nalind) [inline storage/store.go]: It should be fine if it's not set, but now that you mention it, this doesn't seem necessary to make this case work.  I'll drop it.

**Comment by rhatdan** [comment]
:+1:

**Comment by nalind** [comment]
Thanks, merging.

**Pull request merged**